source/parser/riscv/riscv-parser.cpp

```diff
@@ -225,11 +214,10 @@ const SyntaxInformation RiscvParser::getSyntaxInformation() {
   info.addSyntaxRegex("^\\s*\\w+:", SyntaxInformation::Token::Label);
 
   // Add immediate regex
-  // Matches arithmetic expressions containing digits, operators, brackets and
-  // spaces. Expressions need to start with a digit, an open bracket or an unary
-  // operator.
+  // Matches expressions containing digits, letters, operators, brackets and
+  // spaces.
   info.addSyntaxRegex(
-      R"(\b[\+\-0-9\(!~][0-9a-fA-Fx\+\-%\*\/\(\)\|\^&=!<>~\t ]*)",
+      R"([0-9a-zA-Z\+\-%\*\/\(\)\|\^&=!<>~_]+)",
```

Comment:
Maybe making labels/constants (or all "words") highlighted differently in here? (no required, but it would be nice in my opinion)
Also, maybe the constant numbers in a different color? Or is this too late to change?